### ISSUE TITLE
Remote - SSH extension timing out

### ISSUE THREAD

**Issue by carvalhais** [microsoft/vscode-remote-release#3615]
Hi, I'm having issues using the Remote-SSH extension. It looks like I'm able to successfully connect to the remote, but eventually the extension fails (does not open the directory selection dialog), with the following errors:

_Could not fetch remote environment.
Failed to connect to the remote extension host server (Error: Time limit reached)._

Local computer is Windows 10 64bit, and remote is WSL2 with Ubuntu 20.04 (using the host's OpenSSH and bash as default shell, so it will spawn a wsl process on connection).

Bellow is the output from the extension in my local vscode instance:

```
[22:36:51.036] Log Level: 2
[22:36:51.042] remote-ssh@0.51.0
[22:36:51.043] win32 x64
[22:36:51.044] SSH Resolver called for "ssh-remote+niagara", attempt 1
[22:36:51.045] SSH Resolver called for host: niagara
[22:36:51.045] Setting up SSH remote "niagara"
[22:36:51.065] Using commit id "a0479759d6e9ea56afa657e454193f72aef85bd0" and quality "stable" for server
[22:36:51.066] Install and start server if needed
[22:36:51.072] Checking ssh with "ssh -V"
[22:36:51.108] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[22:36:51.110] Running script with connection command: ssh -T -D 50739 niagara bash
[22:36:51.113] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[22:36:51.431] > andre@niagara's password:]0;C:\WINDOWS\System32\cmd.exe
[22:36:51.432] Got some output, clearing connection timeout
[22:36:51.434] Showing password prompt
[22:36:58.401] Got password response
[22:36:58.402] "install" wrote data to terminal: "***************"
[22:36:58.417] > 
[22:36:58.566] > ac9bd6b7c862: running
[22:36:58.585] > Acquiring lock on /home/carvalhais/.vscode-server/bin/a0479759d6e9ea56afa657e454
> 193f72aef85bd0/vscode-remote-lock.carvalhais.a0479759d6e9ea56afa657e454193f72aef
> 85bd0
> \ln /home/carvalhais/.vscode-server/bin/a0479759d6e9ea56afa657e454193f72aef85bd0
> /vscode-remote-lock.carvalhais.a0479759d6e9ea56afa657e454193f72aef85bd0.target /
> home/carvalhais/.vscode-server/bin/a0479759d6e9ea56afa657e454193f72aef85bd0/vsco
> de-remote-lock.carvalhais.a0479759d6e9ea56afa657e454193f72aef85bd0
> Found existing installation at /home/carvalhais/.vscode-server/bin/a0479759d6e9e
> a56afa657e454193f72aef85bd0...
[22:36:58.599] > your 131072x1 screen size is bogus. expect trouble
> Found running server...
> 
> *
> * Reminder: You may only use this software with Visual Studio family products, 
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
> 
> Checking server status on port 34609 with wget
> ac9bd6b7c862: start
> sshAuthSock====
> agentPort==34609==
> osReleaseId==ubuntu==
> arch==x86_64==
> webUiAccessToken====
> tmpDir==/tmp==
> platform==linux==
[22:36:58.606] > 
> ac9bd6b7c862: end
[22:36:58.607] Received install output: 
sshAuthSock====
agentPort==34609==
osReleaseId==ubuntu==
arch==x86_64==
webUiAccessToken====
tmpDir==/tmp==
platform==linux==

[22:36:58.609] Remote server is listening on port 34609
[22:36:58.610] Parsed server configuration: {"agentPort":34609,"osReleaseId":"ubuntu","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","tmpDir":"/tmp","platform":"linux"}
[22:36:58.614] Starting forwarding server. localPort 50742 -> socksPort 50739 -> remotePort 34609
[22:36:58.616] Forwarding server listening on 50742
[22:36:58.617] Waiting for ssh tunnel to be ready
[22:36:58.620] [Forwarding server 50742] Got connection 0
[22:36:58.623] Tunneled remote port 34609 to local port 50742
[22:36:58.624] Resolved "ssh-remote+niagara" to "127.0.0.1:50742"
[22:36:58.646] ------




[22:36:58.674] [Forwarding server 50742] Got connection 1
[22:36:58.675] [Forwarding server 50742] Got connection 2
```

**Comment by roblourens** [comment]
@aeschli I see other instances of this error with the server on WSL, do you know what's going on or is there a dupe issue?

**Comment by aeschli** [comment]
When the resolver returns a address that can not be connected to you get  'Could not fetch remote environment....'.

In WSL I had a bug where I detected WSL1/WSL2 wrong what resulted in the wrong ip to be returned. That I fixed in the meantime. Did you see other issues like that?

**Comment by carvalhais** [comment]
> When the resolver returns a address that can not be connected to you get 'Could not fetch remote environment....'.

I honestly don't believe this is related to a name resolving issue. I had tried removing my `.vscode-server` directory in WSL, which is again installed when I connect from the client vscode. Also I believe the log shows connection is established (even though that's not completely clear to me).

> In WSL I had a bug where I detected WSL1/WSL2 wrong what resulted in the wrong ip to be returned. That I fixed in the meantime. Did you see other issues like that?

Again, I don't think this is related to not being able to connect.

**Comment by roblourens** [comment]
> Also I believe the log shows connection is established (even though that's not completely clear to me).

It doesn't, it only shows that we started the server.

Have you used Remote-WSL with this? 

If you run the command "Kill VS Code Server on Host" then try to connect again with SSH, does it work?

**Comment by carvalhais** [comment]
tl;dr
Try rebooting your machine before embarrassing yourself in GitHub.

It turns out that I was probably doing something stupid that was preventing my client vscode instance from connecting to the server vscode instance, and while troubleshooting as @roblourens suggested I hit a stage where things started working again. I'll be leaving a comment on my issues so that if other's get to this thread they know what worked for me.

**Environment**
I have a vscode client on a desktop running Windows 10 64-bit with both WSL1 and WSL2 which I use while at home because of the convenience of it's dual monitor setup. The desktop vscode works with it's local WSL instances without problems using the Remote-WSL extension.

I also have a vscode server on a laptop running WIndows 10 64-bit with both WSL1 and WSL2 (the vscode server instance runs on the WSL2), on where I want all my dev files to reside because of the convenience of having all my dev environment with me by just closing the lid. It also has vscode that works with it's local WSL instances without problems using the Remote-WSL extension.

The client on the desktop was having issues connecting to the server on the laptop through the LAN using the Remote-SSH extension.

**Potential problems**
While troubleshooting as suggested, I realized I had a bunch of vscode-server processes spawn on the WSL2 environment on the laptop, so I began killing all of them, and at some point I decided to just reboot the machine.

Since WSL2 networking is a lot more involved than WSL1 and requires virtual network setup, port opening on firewalls, and dealing with ever changing IP addresses, I just went the obvious lazy way of circumventing it completely by running the OpenSSH server on the host, not on WSL2 itself, but setting my default shell to WSL2's bash, with the result being I have the best of both worlds.

So when I went to start the OpenSSH service on the Windows host to allow for the client to connect after the reboot, I remembered that perhaps, maybe, it possibly had a slight chance (cough, cough) that I let an OpenSSH server instance running on the WSL2 VM as well, which might have messed things up.

With that said, and jokes apart, before rebooting I did try to stop the WSL sshd service, with no result. But since it was obvious my WSL2 was in a messed up state, the reboot simply solved it.

I apologize for taking your valuable time with this non-issue, I'll be punishing myself in shame for the next half century.

**Comment by roblourens** [comment]
No worries, I'm glad you figured it out!
